samtranslator/model/sam_resources.py

```diff
@@ -839,8 +846,49 @@ def to_cloudformation(self, **kwargs):
         synthetics_canary = self._construct_synthetics_canary()
         resources.append(synthetics_canary)
 
+        # A S3 Bucket resource will be added to the transformed template if the user doesn't provide an artifact
+        # bucket to store canary results
+        if synthetics_canary.ArtifactS3Location is None:
+            s3bucket = self._construct_artifact_bucket()
+            resources.append(s3bucket)
+            synthetics_canary.ArtifactS3Location = {"Fn::Join": ["", ["s3://", {"Ref": s3bucket.logical_id}]]}
+
         return resources
 
+    def _construct_artifact_bucket(self):
+        """Constructs a S3Bucket resource to store canary artifacts.
+
+        :returns: the generated S3Bucket
+        :rtype: model.s3.S3Bucket
+        """
+
+        # Construct the LogicalId of S3Bucket by appending ArtifactBucket to the Canary LogicalId. Once deployed, the
+        # bucket name will be automatically generated by Cloudformation.
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-name.html
+
+        logical_id = self.logical_id + "ArtifactBucket"
```

Comment:
When we append string to a logical ID, do we need to worry about char limit?

Reply:
the char limit for logical id of a resource is 255 characters so it's pretty big. Also, in Serverless Function when a role is being created the logical id for that role is self.logical_id + "Role" and there is no check for char limit so i didn't add a check here. Think there is no check because CFN handles it when deploying.

Reply:
Fair enough if we don't check in current code. Thanks for checking